### PR TITLE
detect_arch in ClusterDescriptor

### PR DESCRIPTION
**Pull request by broskoTT**
### Issue
Related to #99 
Aftermath of #175 

### Description
Make detect_arch available through cluster_descriptor which should be the only real endpoint when reading information about chips in the cluster.

### List of the changes
- Removed detect_arch from cluster.h
- Added arch map in ClusterDescriptor
- Added static tt_ClusterDescriptor::detect_arch which should be used instead of previously defined detect_arch

### Testing
Wrote a test which verifies this works as intended

### API Changes
This PR has API changes:
- [x] tt_metal approved PR pointing to this branch: https://github.com/tenstorrent/tt-metal/pull/15564
- [x] tt_debuda approved PR pointing to this branch: https://github.com/tenstorrent/tt-lens/pull/201

### CODE REVIEW

File: device/tt_cluster_descriptor.cpp [hunk 1]
```diff
@@ -856,8 +872,21 @@ int tt_ClusterDescriptor::get_ethernet_link_distance(chip_id_t chip_a, chip_id_t
 }
 
 BoardType tt_ClusterDescriptor::get_board_type(chip_id_t chip_id) const {
-    BoardType board_type = this->chip_board_type.at(chip_id);
-    return board_type;
+    if (chip_board_type.find(chip_id) == chip_board_type.end()) {
+        return BoardType::DEFAULT;
```
[the remark this answers is not in the document]
Comment (broskoTT) [reply device/tt_cluster_descriptor.cpp:876]: Having null in cluster descriptor is posible, and I think we need to cover that case.
Maybe I should just change DEFAULT to INVALID in the enum to better convey this? Or UNKNOWN ?

[the remark this answers is not in the document]
Comment (pjanevskiTT) [reply device/tt_cluster_descriptor.cpp:876]: Yes I think INVALID or UNKNOWN would suit this case much better, DEFAULT doesn't make much sense to me. So if somethings has null, it won't be in this `chip_board_type` map. I was reading this code such that you were returning default if chip_id wasn't in the map?

File: device/tt_cluster_descriptor.cpp [hunk 2]
```diff
@@ -781,6 +784,19 @@ void tt_ClusterDescriptor::fill_chips_grouped_by_closest_mmio() {
     }
 }
 
+tt::ARCH tt_ClusterDescriptor::arch_from_string(std::string arch_str) {
```
[the remark this answers is not in the document]
Comment (broskoTT) [reply device/tt_cluster_descriptor.cpp:787]: I'll solve this in another PR, there are a couple of stuff to resolve around tt::ARCH.
Will close this comment when I reference the PR

File: device/tt_cluster_descriptor.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (broskoTT) [reply device/tt_cluster_descriptor.cpp:876]: Ah right, your comment was that this should be an assert.

[the remark this answers is not in the document]
Comment (broskoTT) [reply device/tt_cluster_descriptor.cpp]: Added log_assert, and changed default to unknown

File: device/tt_cluster_descriptor.cpp [hunk 3]
```diff
@@ -856,8 +872,21 @@ int tt_ClusterDescriptor::get_ethernet_link_distance(chip_id_t chip_a, chip_id_t
...
-    return board_type;
+    if (chip_board_type.find(chip_id) == chip_board_type.end()) {
+        return BoardType::DEFAULT;
+    }
+    return chip_board_type.at(chip_id);
+}
+
+tt::ARCH tt_ClusterDescriptor::get_arch(chip_id_t chip_id) const {
+    if (chip_arch.find(chip_id) == chip_arch.end()) {
+        return tt::ARCH::Invalid;
```
[the remark this answers is not in the document]
Comment (broskoTT) [reply device/tt_cluster_descriptor.cpp]: added log_assert

File: device/tt_cluster_descriptor.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (pjanevskiTT) [reply device/tt_cluster_descriptor.cpp:787]: makes sense

**Review by pjanevskiTT — APPROVED** [review]
thanks for the changes

File: device/tt_cluster_descriptor.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (broskoTT) [reply device/tt_cluster_descriptor.cpp:787]: I've left a comment in https://github.com/tenstorrent/tt-umd/pull/357 to address this.
I'll merge this one first.

**Pull request merged**